### ISSUE TITLE
Invalid escape sequences in command line

### ISSUE THREAD

**Issue by pcittadini** [coreos/fleet#1367]
After switching to a brand new stack with etcd2, deploying this (JUST this ) unit file (a node of an ES cluster):
```
[Unit]
After=docker.service
Requires=docker.service
Requires=elasticsearch-data@%i.service
Requires=elasticsearch-discovery@%i.service

[Service]
User=core
TimeoutSec=0
Restart=always
RestartSec=1m
EnvironmentFile=/etc/environment
SuccessExitStatus=255
ExecStartPre=-/usr/bin/docker kill %p-%i
ExecStartPre=-/usr/bin/docker rm %p-%i
ExecStartPre=/usr/bin/docker pull quay.io/myprivaterepo/elasticsearch
# register with skydns
ExecStartPre=/usr/bin/etcdctl set /skydns/local/skydns/services/elasticsearch/elasticsearch-%i '{"host":"${COREOS_PRIVATE_IPV4}","priority":20}'
ExecStart=/bin/bash -c 'curl -f ${COREOS_PRIVATE_IPV4}:2379/v2/keys/services/elasticsearch; if [ "$?" = "0" ]; then UNICAST_HOSTS=$(etcdctl ls --recursive /services/elasticsearch | sed "s/\/services\/elasticsearch\///g"| sed "s/$/:9300/"| paste -s -d","); else UNICAST_HOSTS=""; fi; /usr/bin/docker run --rm --name %p-%i --publish 9200:9200 --publish 9300:9300 --volumes-from elasticsearch-data-%i -e ELASTICSEARCH_CLUSTER_NAME=logstash quay.io/myprivaterepo/elasticsearch -Des.node.name=%p-%i -Des.network.publish_host=${COREOS_PRIVATE_IPV4} -Des.discovery.zen.ping.multicast.enabled=false -Des.discovery.zen.ping.unicast.hosts=$UNICAST_HOSTS'
ExecStop=/usr/bin/docker stop %p-%i
ExecStopPost=-/usr/bin/docker rm %p-%i
# deregister with skydns
ExecStopPost=/usr/bin/etcdctl rm --recursive /skydns/local/skydns/services/elasticsearch/

[X-Fleet]
Conflicts=%p@*.service
```
Will result in an etcd shutdown on several nodes, and the journal available for few seconds before the whole cluster crashes, just repost this:
```
Sep 30 08:33:36 ip-10-128-0-92 systemd[1]: [/run/fleet/units/elasticsearch@1.service:20] Invalid escape sequences in command line: "/bin/bash -c 'curl -f ${COREOS_PRIVATE_IPV4}:2379/v2/keys/services/elasticsearch; if [ "$?" = "0" ]; then UNICAST_HOSTS=$(etcdctl ls --recursive /services/elasticsearch | sed "s/\/services\/elasticsearch\///g"| sed "s/$/:9300/"| paste -s -d","); else UNICAST_HOSTS=""; fi; /usr/bin/docker run --rm --name %p-%i --publish 9200:9200 --publish 9300:9300 --volumes-from elasticsearch-data-%i -e ELASTICSEARCH_CLUSTER_NAME=logstash quay.io/myprivaterepo/elasticsearch -Des.node.name=%p-%i -Des.network.publish_host=${COREOS_PRIVATE_IPV4} -Des.discovery.zen.ping.multicast.enabled=false -Des.discovery.zen.ping.unicast.hosts=$UNICAST_HOSTS'"
Sep 30 08:33:39 ip-10-128-0-92 systemd[1]: Starting Elasticsearch Service...
Sep 30 08:33:40 ip-10-128-0-92 docker[1340]: Pulling repository quay.io/myprivaterepo/elasticsearch
Sep 30 08:33:41 ip-10-128-0-92 systemd[1]: elasticsearch@1.service: Control process exited, code=exited status=2
Sep 30 08:33:41 ip-10-128-0-92 systemd[1]: elasticsearch@1.service: Control process exited, code=exited status=4
Sep 30 08:33:41 ip-10-128-0-92 systemd[1]: Stopped elasticsearch@1.service.
Sep 30 08:33:41 ip-10-128-0-92 systemd[1]: elasticsearch@1.service: Unit entered failed state.
Sep 30 08:33:41 ip-10-128-0-92 systemd[1]: elasticsearch@1.service: Failed with result 'exit-code'
```
The very same unit file is deployed on a cluster running on etcd 0.5, without trouble. 
The most important problem is not to sort out why this can't get deployed, but how a single unit file can break a brand new cluster.

**Comment by jonboulle** [comment]
What version of fleet is running on the two clusters?

**Comment by mindnuts** [comment]
@jonboulle Same issue with my setup. In my case etcd does not crash.

CoreOS alpha (815.0.0)
fleetd version 0.11.5
etcd version 0.4.9


My Unit file:
```
[Unit]
Description=Elasticsearch Service
Documentation=http://www.elasticsearch.org/guide/en/elasticsearch/reference/current/index.html
After=docker.service
Requires=docker.service
Requires=elasticsearch-data@%i.service
Requires=elasticsearch-discovery@%i.service

[Service]
TimeoutSec=0
Restart=always
RestartSec=1m
EnvironmentFile=/etc/environment
SuccessExitStatus=255

ExecStartPre=-/usr/bin/docker kill %p-%i
ExecStartPre=-/usr/bin/docker rm %p-%i
ExecStartPre=/usr/bin/docker pull digitalwonderland/elasticsearch

ExecStart=/bin/bash -c '\
  curl -f ${COREOS_PRIVATE_IPV4}:4001/v2/keys/services/elasticsearch; \
  if [ "$?" = "0" ]; then \
      UNICAST_HOSTS=$(etcdctl ls --recursive /services/elasticsearch \
                      | sed "s/\/services\/elasticsearch\///g" \
                      | sed "s/$/:9300/" \
                      | paste -s -d","); \
  else \
      UNICAST_HOSTS=""; \
  fi; \
  /usr/bin/docker run \
    --rm \
    --name %p-%i \
    --publish 9200:9200 \
    --publish 9300:9300 \
    --volumes-from elasticsearch-data-%i \
    -e ELASTICSEARCH_CLUSTER_NAME=logstash \
    digitalwonderland/elasticsearch \
    -Des.node.name=%p-%i \
    -Des.network.publish_host=${COREOS_PRIVATE_IPV4} \
    -Des.discovery.zen.ping.multicast.enabled=false \
    -Des.discovery.zen.ping.unicast.hosts=$UNICAST_HOSTS'
ExecStop=/usr/bin/docker stop %p-%i
ExecStopPost=-/usr/bin/docker rm %p-%i

[X-Fleet]
Conflicts=%p@*.service
```

Here is the error found in Journal:

```
Oct 05 08:15:03 core-01 systemd[1]: [/run/fleet/units/elasticsearch@3.service:18] Invalid escape sequences in command line: "/bin/bash -c '   curl -f ${COREOS_PRIVATE_IPV4}:4001/v2/keys/services/elasticsearch;    if [ "$?" = "0" ]; then        UNICAST_HOSTS=$(etcdctl ls --recursive /services/elasticsearch                        | sed "s/\/services\/elasticsearch\///g"                        | sed "s/$/:9300/"                        | paste -s -d",");    else        UNICAST_HOSTS="";    fi;    /usr/bin/docker run      --rm      --name %p-%i      --publish 9200:9200      --publish 9300:9300      --volumes-from elasticsearch-data-%i      -e ELASTICSEARCH_CLUSTER_NAME=logstash      digitalwonderland/elasticsearch      -Des.node.name=%p-%i      -Des.network.publish_host=${COREOS_PRIVATE_IPV4}      -Des.discovery.zen.ping.multicast.enabled=false      -Des.discovery.zen.ping.unicast.hosts=$UNICAST_HOSTS'"

```

**Comment by mindnuts** [comment]
Never mind got it working finally.

TLDR; escape $ with $$, \ as \\

Unit file for reference:
```
[Unit]
Description=Elasticsearch Service
Documentation=http://www.elasticsearch.org/guide/en/elasticsearch/reference/current/index.html
After=docker.service
Requires=docker.service
Requires=elasticsearch-data@%i.service
Requires=elasticsearch-intelligent-discovery@%i.service

[Service]
TimeoutSec=0
Restart=always
RestartSec=1m
EnvironmentFile=/etc/environment
SuccessExitStatus=255

ExecStartPre=-/usr/bin/docker kill %p-%i
ExecStartPre=-/usr/bin/docker rm %p-%i
ExecStartPre=/usr/bin/docker pull digitalwonderland/elasticsearch
ExecStart=/bin/bash -c 'curl -f ${COREOS_PRIVATE_IPV4}:4001/v2/keys/services/elasticsearch; \
  if [ "$$?" = "0" ]; then \
      UNICAST_HOSTS=$$(etcdctl ls --recursive /services/elasticsearch | sed "s/\\/services\\/elasticsearch\\///g" | sed "s/$$/:9300/" | paste -s -d","); \
  else \
      UNICAST_HOSTS=""; \
  fi; \
  /usr/bin/docker run \
    --rm \
    --name %p-%i \
    --publish 9200:9200 \
    --publish 9300:9300 \
    --volumes-from elasticsearch-data-%i \
    -e ELASTICSEARCH_CLUSTER_NAME=logstash \
    digitalwonderland/elasticsearch \
    --node.name=%p-%i \
    --network.publish_host=${COREOS_PRIVATE_IPV4} \
    --discovery.zen.ping.multicast.enabled=false \
    --discovery.zen.ping.unicast.hosts=${UNICAST_HOSTS}'

ExecStop=/usr/bin/docker stop %p-%i
ExecStopPost=-/usr/bin/docker rm %p-%i

[X-Fleet]
Conflicts=%p@*.service
```

**Comment by pcittadini** [comment]
Got it working to, escaped dirty characters in file.
For the Etcd2 crash, in the dependency unit file there were a wrong Requires clause, pointing to etcd.service instead of etcd2.service, this caused a major crash. 
Closing
